machine/xen/v1alpha1.go

```diff
@@ -0,0 +1,536 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file except in compliance with the License.
+
+//go:build xen
+// +build xen
+
+package xen
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+	"time"
+
+	zip "api.zip"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+
+	machinev1alpha1 "kraftkit.sh/api/machine/v1alpha1"
+	"kraftkit.sh/config"
+	"kraftkit.sh/internal/logtail"
+	"kraftkit.sh/log"
+	"kraftkit.sh/machine/network/macaddr"
+	"kraftkit.sh/unikraft/export/v0/ukargparse"
+	"kraftkit.sh/unikraft/export/v0/uknetdev"
+	"kraftkit.sh/unikraft/export/v0/vfscore"
+	"xenbits.xenproject.org/git-http/xen.git/tools/golang/xenlight"
+)
+
+const (
+	XenMemoryScale   = 1024
+	XenMemoryDefault = 64
+	XenCPUsDefault   = 1
+)
+
+type machineV1alpha1Service struct{}
+
+func NewMachineV1alpha1Service(ctx context.Context) (machinev1alpha1.MachineService, error) {
+	return &machineV1alpha1Service{}, nil
+}
+
+func (service *machineV1alpha1Service) Create(ctx context.Context, machine *machinev1alpha1.Machine) (*machinev1alpha1.Machine, error) {
+	cfg, err := NewXenConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	if machine.Status.KernelPath == "" {
+		return machine, fmt.Errorf("cannot create xen instace without a kernel")
+	}
+
+	if !slices.Contains([]string{"arm64", "x86_64", "amd64", "arm"}, machine.Spec.Architecture) {
```

Comment:
I think you can get this list from a function in kraftkit, try to look for it I can't remember